pkg/build/gobuild.go

```diff
@@ -86,19 +86,41 @@ func NewGo(options ...Option) (Interface, error) {
 	return gbo.Open()
 }
 
+// findImportPath uses the go list command to find the full import path from a package.
+func (g *gobuild) findImportPath(importPath, format string) (string, error) {
```

Comment:
Can these calls be stored in a cache and then flushed each time a file change is detected? This will likely provide an enormous speedup.

Reply:
Maybe -- we still have to call that once per unique string in the input yaml, so this would help a bit but not _that_ much.

Reply:
Hmmm - could calling go list once at the top level work? That way you could avoid having to deal with lots of go list calls.

Reply:
We could do this in batch, go list accepts multiples packages as argument, would need to try this out. @jonjohnsonjr can you share to me how to reproduce the ko resolve from knative that you tried?

Reply:
I'm also trying to figure someway to prune some of these unique strings on the input yaml, because not every string are actually valid packages names right?

Reply:
> can you share to me how to reproduce the ko resolve from knative that you tried?

Clone https://github.com/knative/serving and run `ko resolve -f config/`.

> not every string are actually valid packages names right

Yeah we could do some filtering, but I think most strings are going to be potentially valid packages.

Reply:
I think I got it to a time that is fine. The biggest culprit for this slow down was that the method `isSupportedReference` was being called a lot of times.

I have done two things: 
- Pruned some invalid package names (found a way based on the internal load packing from the Go command);
- Cached the result of this method for each import path;

We sure could try to batch `go list` calls, but that would make it require a lot more effort and I think that should be done in the future. 

Also I have some doubts of which strings we do try to build and replace on the YAML, currently we try check every string from the YAML, such as: normal strings, map keys, map values and array, do we actually need them all?

Reply:
@jonjohnsonjr I tested out on the knative's serving repo and it was down to 35~40 seconds on my machine, can you run this test on your side as well?

Reply:
Friendly ping @jonjohnsonjr.
Also @clrprod as you pointed out to cache the results.